### ISSUE TITLE
`no-conditional-expect` update to take `expect.assertions` into account

### ISSUE THREAD

**Issue by vidarc** [vitest-dev/eslint-plugin-vitest#837]
Currently the following will fail `no-conditional-expect`

```javascript
test('some test', () => {
  expect.assertions(1);

  if (someValue) {
    expect(someValue).toStrictEqual(123);
  }
});
```

But I would argue that this should pass, or be an option to allow it to pass.

Admittedly, I currently am using conditional expressions to make the TypeScript compiler happy, for example

```typescript
test('some test with types', async () => {
  expect.assertions(1);

  try {
    await someFunction();
  } catch (error) {
    if (error instanceof MyCustomError) {
      expect(error.myCustomProp).toStrictEqual('something');
    }
  }
});
```
but do think this would be overall a good addition if others agree

**Comment by veritem** [comment]
@y-hsgw I agree, @vidarc can we have this behind a user option instead of it being a default behavior?
